politeiad/backendv2/tstorebe/plugins/pi/cmds.go

```diff
@@ -165,6 +169,192 @@ func tokenMatches(cmdToken []byte, payloadToken string) error {
 	return nil
 }
 
+// cmdSummary returns the pi summary of a proposal.
+func (p *piPlugin) cmdSummary(token []byte) (string, error) {
+	// Get record metadata
+	r, err := p.recordAbridged(token)
+	if err != nil {
+		return "", err
+	}
+	mdState := r.RecordMetadata.State
+	mdStatus := r.RecordMetadata.Status
+	voteStatus := ticketvote.VoteStatusInvalid
+	var bsc *pi.BillingStatusChange
+
+	// Fetch vote status & billing status change if they are needed in order
+	// to determine the proposal status.
+	if mdState == backend.StateVetted {
+		// If proposal status is public fetch vote status.
+		if mdStatus == backend.StatusPublic {
+			vs, err := p.voteSummary(token)
+			if err != nil {
+				return "", err
+			}
+			voteStatus = vs.Status
+			// If vote status is approved fetch billing status change.
+			if voteStatus == ticketvote.VoteStatusApproved {
+				bsc, err = p.billingStatusChange(token)
+				if err != nil {
+					return "", err
+				}
+			}
+		}
+	}
+
+	proposalStatus, err := p.proposalStatus(mdState, mdStatus, voteStatus, bsc)
```

Comment:
A nice things about fetching the vote status and billing status before calling `proposalStatus` is that now you can add tests to `proposalStatus`.  

Typically I like to test API as input and output with the implementation as a black box, but we still don't have a testing framework setup to make the vote status and billing status calls yet.  That's ok for now.  Add test coverage to `proposalStatus` then when we do get the testing framework setup we can transfer those test cases over to strictly API input/output.

Reply:
good point